src/system/BlocksModule/Controller/BlockController.php

```diff
@@ -0,0 +1,286 @@
+<?php
+/**
+ * Copyright Zikula Foundation 2015 - Zikula Application Framework
+ *
+ * This work is contributed to the Zikula Foundation under one or more
+ * Contributor Agreements and licensed to You under the following license:
+ *
+ * @license GNU/LGPLv3 (or at your option, any later version).
+ *
+ * Please see the NOTICE file distributed with this source code for further
+ * information regarding copyright and licensing.
+ */
+
+namespace Zikula\BlocksModule\Controller;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Zikula\BlocksModule\Api\BlockApi;
+use Zikula\BlocksModule\Entity\BlockEntity;
+use Zikula\BlocksModule\Form\Type\BlockType;
+use Zikula\Core\BlockControllerInterface;
+use Zikula\Core\Controller\AbstractController;
+use Zikula\Core\Response\Ajax\FatalResponse;
+use Zikula\Core\Response\Ajax\ForbiddenResponse;
+use Zikula\Core\Theme\Annotation\Theme;
+
+/**
+ * Class BlockController
+ * @package Zikula\BlocksModule\Controller
+ * @Route("/admin/block")
+ */
+class BlockController extends AbstractController
+{
+    /**
+     * @Route("/new")
+     * @Theme("admin")
+     * @param Request $request
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
+     */
+    public function newAction(Request $request)
+    {
+        $form = $this->createFormBuilder()
+            ->add('bkey', 'Symfony\Component\Form\Extension\Core\Type\ChoiceType', [
+                'placeholder' => 'Choose a block type',
+                'choices' => $this->get('zikula_blocks_module.api.block')->getAvailableBlockTypes(),
+                'label' => 'Block type',
+            ])
+            ->add('choose', 'Symfony\Component\Form\Extension\Core\Type\SubmitType')
+            ->getForm();
+        $form->handleRequest($request);
+        if ($form->isValid()) {
+            $bkey = json_encode($form->getData()['bkey']);
+
+            return $this->redirect($this->generateUrl('zikulablocksmodule_block_edit', ['bkey' => $bkey]));
+        }
+
+        return $this->render('ZikulaBlocksModule:Admin:new.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * Create a new block or edit an existing block.
+     *
+     * @Route("/edit/{blockEntity}", requirements={"blockEntity" = "^[1-9]\d*$"})
```

Comment:
you do. the `^` means that the first character must be 1-9 and the `*`means repeated 0 to infinite times until the `$` end (means it must end with a 1-9)